playbook.yaml

```diff
@@ -135,3 +135,59 @@
         merge_type:
           - merge
       with_filetree: objects/patches/
+
+    - name: Get Application Controller Service Account
+      k8s_info:
+        kubeconfig: "{{ kubeconfig }}"
+        kind: ServiceAccount
+        name: argocd-application-controller
+        namespace: "{{ namespace }}"
+        verify_ssl: "no"
+      register: appctrl_sa
+
+    - name: Get Application Controller Service Account Secrets
+      k8s_info:
+        kubeconfig: "{{ kubeconfig }}"
+        kind: Secret
+        name: "{{ item.name }}"
+        namespace: "{{ namespace }}"
+        verify_ssl: "no"
+      register: appctrl_sa_secrets
+      when: "'token' in item.name"
+      with_items: "{{ appctrl_sa.resources[0].secrets }}"
+
+    - set_fact:
+        appctrl_sa_token: "{{ item.resources[0].data.token | b64decode }}"
+      when: "'resources' in item"
+      with_items: "{{ appctrl_sa_secrets.results }}"
+      no_log: "True"
+
+    - name: Deploy ArgoCD Projects
```

Comment:
Does this create the OpenShift projects? When I’ve tried to do this in the past against production PSI openshift clusters I’ve hit permissions issues. Does this actually work?

Reply:
Resolved. I thought these were openshift namespaces, not argocd projects.